keyboards/cutie_club/fidelity/keymaps/via/keymap.c

```diff
@@ -0,0 +1,49 @@
+        KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,      KC_TRNS,    KC_TRNS,          KC_TRNS,
+        KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,      KC_TRNS,    KC_TRNS,          KC_TRNS,
+        KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,      KC_TRNS,             KC_TRNS, KC_TRNS,
+        KC_TRNS,                  KC_TRNS,                                                         KC_TRNS,                                                          KC_TRNS,    KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+    [2] = LAYOUT(
+        KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,      KC_TRNS,    KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,      KC_TRNS,    KC_TRNS,          KC_TRNS,
+        KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,      KC_TRNS,    KC_TRNS,          KC_TRNS,
+        KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,      KC_TRNS,             KC_TRNS, KC_TRNS,
+        KC_TRNS,                  KC_TRNS,                                                         KC_TRNS,                                                          KC_TRNS,    KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+    [3] = LAYOUT(
+        KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,      KC_TRNS,    KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,      KC_TRNS,    KC_TRNS,          KC_TRNS,
+        KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,      KC_TRNS,    KC_TRNS,          KC_TRNS,
+        KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,      KC_TRNS,             KC_TRNS, KC_TRNS,
+        KC_TRNS,                  KC_TRNS,                                                         KC_TRNS,                                                          KC_TRNS,    KC_TRNS, KC_TRNS, KC_TRNS
+    )
```

Comment:
```suggestion
    )
```

Reply:
I was under the impression that, VIA keymaps should pre-allocate the number of dynamic layers the keyboard wants to support, so if I want to allow users 4 layers in VIA I would need 4 layers allocatied in the VIA keymap.

It is very likely I am misunderstanding, I just want to check?

Reply:
> VIA keymaps should pre-allocate the number of dynamic layers the keyboard wants to support,

This is controlled with `DYNAMIC_KEYMAP_LAYER_COUNT`, not via the method you have attempted to use.

Padding the keymap array for init purposes is also no longer required, as unspecified layers will default to `KC_TRANS`.

Reply:
Gotya, that makes sense, thank you for the explination (o˘◡˘o)